foreman/data_refinery_foreman/surveyor/array_express.py

```diff
@@ -16,68 +17,111 @@
-
-            for data_file in data_files:
-                if (data_file["kind"] == "raw"):
-                    url = data_file["url"].replace("\\", "")
-                    # This is another place where this is still a POC.
-                    # More work will need to be done to determine some
-                    # of these additional metadata fields.
-                    self.handle_batch(Batch(size_in_bytes=data_file["size"],
-                                            download_url=url,
-                                            raw_format="MICRO_ARRAY",
-                                            processed_format="PCL",
-                                            accession_code=accession_code,
-                                            organism=1))
-
-        return True
+    @staticmethod
+    def get_experiment_metadata(experiment_accession_code):
+        experiment_request = requests.get(EXPERIMENTS_URL + experiment_accession_code)
+        parsed_json = experiment_request.json()["experiments"]["experiment"][0]
+
+        experiment = {}
+
+        experiment["name"] = parsed_json["name"]
+        experiment["experiment_accession_code"] = experiment_accession_code
+
+        # If there is more than one arraydesign listed in the experiment
+        # then there is no other way to determine which array was used
+        # for which sample other than looking at the header of the CEL
+        # file. That obviously cannot happen until the CEL file has been
+        # downloaded so we can just mark it as UNKNOWN and let the
+        # downloader inspect the downloaded file to determine the
+        # array then.
+        if len(parsed_json["arraydesign"]) == 0:
+            logger.warn("Experiment %s has no arraydesign listed.", experiment_accession_code)
+            experiment["platform_accession_code"] = "UNKNOWN"
+        elif len(parsed_json["arraydesign"]) > 1:
+            experiment["platform_accession_code"] = "UNKNOWN"
+        else:
+            experiment["platform_accession_code"] = \
+                parsed_json["arraydesign"][0]["accession"]
+
+        experiment["release_date"] = parsed_json["releasedate"]
+
+        if "lastupdatedate" in parsed_json:
+            experiment["last_update_date"] = parsed_json["lastupdatedate"]
+        else:
+            experiment["last_update_date"] = parsed_json["releasedate"]
+
+        return experiment
+
+    def survey(self):
+        experiment_accession_code = (
+            SurveyJobKeyValue
+            .objects
+            .get(survey_job_id=self.survey_job.id,
+                 key__exact="experiment_accession_code")
+            .value
+        )
+
+        logger.info("Surveying experiment with accession code: %s.", experiment_accession_code)
+
+        experiment = self.get_experiment_metadata(experiment_accession_code)
+
+        r = requests.get(SAMPLES_URL.format(experiment_accession_code))
+        samples = r.json()["experiment"]["sample"]
```

Comment:
My answer here is basically the same as the one above except that I'll add that at some point surveying will be triggered by a foreman who will be responsible for starting and retrying survey_jobs. If the API is down and the job fails then once I get around to writing the foreman code it'll have mechanisms to retry the survey_jobs a few times so hopefully the API will come back up before the retry limit is reached.